test/e2e/performanceprofile/functests/1_performance/irqbalance.go

```diff
@@ -0,0 +1,315 @@
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
+
+			By(fmt.Sprintf("Checking the default IRQ affinity on node %q", node.Name))
+			smpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get default smp affinity")
+
+			By(fmt.Sprintf("Checking the online CPU Set on node %q", node.Name))
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get Online CPUs list")
+
+			// expect no irqbalance run in the system already, AKA start from pristine conditions.
+			// This is not an hard requirement, just the easier state to manage and check
+			Expect(smpAffinitySet.Equals(onlineCPUsSet)).To(BeTrue(), "found default_smp_affinity %v, expected %v - IRQBalance already run?", smpAffinitySet, onlineCPUsSet)
+
+			origBannedCPUsFile := "/etc/sysconfig/orig_irq_banned_cpus"
+			By(fmt.Sprintf("Checking content of %q on node %q", origBannedCPUsFile, node.Name))
+			expectFileEmpty(node, origBannedCPUsFile)
+		})
+
+		It("Should DO overwrite the banned CPU set on CRI-O restart", func() {
+
+			nodeIdx := pickNodeIdx(workerRTNodes)
+			node := &workerRTNodes[nodeIdx]
+			By(fmt.Sprintf("verifying worker node %q", node.Name))
+
+			var err error
+
+			By(fmt.Sprintf("Checking the default IRQ affinity on node %q", node.Name))
+			smpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get default smp affinity")
+
+			By(fmt.Sprintf("Checking the online CPU Set on node %q", node.Name))
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(node)
+			Expect(err).ToNot(HaveOccurred(), "failed to get Online CPUs list")
+
+			// expect no irqbalance run in the system already, AKA start from pristine conditions.
+			// This is not an hard requirement, just the easier state to manage and check
+			Expect(smpAffinitySet.Equals(onlineCPUsSet)).To(BeTrue(), "found default_smp_affinity %v, expected %v - IRQBalance already run?", smpAffinitySet, onlineCPUsSet)
+
+			// setup the CRI-O managed irq banned cpu list
+			By("Preparing fake data for the irqbalance config file")
+			irqBalanceConfFile := "/etc/sysconfig/irqbalance"
+			restoreIRQBalance := makeBackupForFile(node, irqBalanceConfFile)
+			defer restoreIRQBalance()
+
+			// completely fake data. We are backupping the original file anyway, and we succeed if we have empty ban list anyway. So it's good.
+			_, err = nodes.ExecCommandOnNode([]string{"echo", "IRQBALANCE_BANNED_CPUS=2,3", ">", "/rootfs/" + irqBalanceConfFile}, node)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Preparing fake data for the irqbalance cpu ban list file")
+			origBannedCPUsFile := "/etc/sysconfig/orig_irq_banned_cpus"
+			restoreBanned := makeBackupForFile(node, origBannedCPUsFile)
+			defer restoreBanned()
+
+			_, err = nodes.ExecCommandOnNode([]string{"/usr/bin/truncate", "-s", "0", "/rootfs/" + origBannedCPUsFile}, node)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
This should be tested without touching private crio files. Why not follow the same procedure used in the "Verify irqbalance configuration handling" and create a pod that updates the banned cpus and then restart crio? Or add this kind of testcase to crio itself since this is really testing crio behaviour.

Reply:
Because we don't have any control over the content of the `orig_irq_banned_cpus`. So the closest we can do is to create a controlled environment.

Reply:
I guess we can agree to disagree - I'm not an owner so you don't need my approval.